### ISSUE TITLE
Non-mocked requests hang indefinitely

### ISSUE THREAD

**Issue by hmorey3** [mswjs/interceptors#199]
I just started setting up mswjs/interceptors. While I am able to intercept requests and return mocked values just fine, when I removed the mocked response in hopes that it will allow it to call through, the http request hangs indefinitely. When I remove the interceptors altogether, the request resolves just fine. Any thoughts?

```
import { createInterceptor } from '@mswjs/interceptors'
import nodeInterceptors from '@mswjs/interceptors/lib/presets/node'
import got from 'got'

const interceptor = createInterceptor({
  modules: nodeInterceptors,
  resolver(_request, _ref) {}
})
interceptor.apply()

const response = await got.get('url-omitted') // hangs indefinitely
console.log('response', response.body, response.statusCode, response.statusMessage)
```

**Comment by dhruvit-r** [comment]
@hmorey3 I faced this issue too. Here's a hacky solution that works for me. Note that there's other code in the actual file, so think of this as more of a pseudo code.

```ts
import { createInterceptor, IsomorphicRequest } from '@mswjs/interceptors'
import { NodeClientOptions } from '@mswjs/interceptors/lib/interceptors/ClientRequest/NodeClientRequest'
import { NormalizedClientRequestArgs } from '@mswjs/interceptors/lib/interceptors/ClientRequest/utils/normalizeClientRequestArgs'
import nodeInterceptors from '@mswjs/interceptors/lib/presets/node'

type NodeClientRequest = typeof import('@mswjs/interceptors/lib/interceptors/ClientRequest/NodeClientRequest')
const NodeClientRequest: NodeClientRequest = require('@mswjs/interceptors/lib/interceptors/ClientRequest/NodeClientRequest')

class _NodeClientRequest extends NodeClientRequest.NodeClientRequest {
  constructor(args: NormalizedClientRequestArgs, options: NodeClientOptions) {
    const http: typeof import('http') = require('http')
    if (args[1].headers['x-ignore-interceptor']) {
      delete args[1].headers['x-ignore-interceptor']
      super(args, options)
      this.end = http.ClientRequest.prototype.end as any
      this.emit = http.ClientRequest.prototype.emit as any
      this.write = http.ClientRequest.prototype.write as any
      return
    }
    super(args, options)
  }
}

NodeClientRequest.NodeClientRequest = _NodeClientRequest

const interceptor = createInterceptor({
  modules: nodeInterceptors,
  // eslint-disable-next-line @typescript-eslint/no-empty-function
  resolver: () => {},
})
interceptor.apply()
```

To send a request that doesn't need to be intercepted, include the `x-ignore-interceptor` header as part of the request. You could also change the `if (args[1].headers['x-ignore-interceptor'])` check to something else, such as `if (args[1].headers['x-mock-request'])` and add `x-mock-request header: true` to request you actually want to mock.

**Comment by kettanaito** [comment]
Can someone please reproduce this with the latest version of the library?

There's been quite a number of changes to the internal mechanism of the interceptors, which involved public API changes as well. I'd expect this not to be the case anymore. Would someone had the time to prove this is still an issue?

**Comment by kettanaito** [comment]
This has been reported on a rather old version of the library. Please update to the latest version, there's a solid chance it's fixed. 

There's also an automated test using `got` and a bypassed request that confirms this issue doesn't happen. If it still happens to you after the update, please submit a reproduction repository. Thanks.

**Comment by kettanaito** [comment]
## Released: v0.19.3 🎉

This has been released in v0.19.3!

- 📄 [**Release notes**](https://github.com/mswjs/interceptors/releases/tag/v0.19.3)
- 📦 [npm package](https://www.npmjs.com/package/@mswjs/interceptors/v/0.19.3)

Make sure to always update to the latest version (`npm i @mswjs/interceptors@latest`) to get the newest features and bug fixes.

---

_Predictable release automation by [@ossjs/release](https://github.com/ossjs/release)_.
